corehq/middleware.py

```diff
@@ -257,16 +260,24 @@ def __init__(self, get_response=None):
             re.compile(regex.format(domain=legacy_domain_re)) for regex in regexes
         ]
 
-    def _bypass_sessions(self, request, domain):
-        return (toggles.BYPASS_SESSIONS.enabled(domain) and
+    @memoized
```

Comment:
I think this will live for the lifetime of the process since Django doesn't re-create the middleware classes for each requests (I think).

You could rather just save the outcome as an attribute on the request itself like this: https://github.com/dimagi/commcare-hq/blob/cc93bc6ab718bde08e8d16e374d5cc25eff94958/corehq/util/quickcache.py#L34

Reply:
That's what I was intending, though.

Reply:
you want to cache the value forever? but it's only needed for the duration of the request.

Reply:
Ha, sorry. I thought the process is per request not per Django start. I will fix it, thanks for hi-lighting, I didn't know that.

Reply:
Fixed now

Reply:
memoized decorator is still there
```suggestion
```

Reply:
Fixed